File: connectorSamples/xyz-earthquake/src/storage/StatisticsResponse.js
const XyzResponse = require("./XyzResponse");

class StatisticsResponse extends XyzResponse {

	constructor() {
		super();
		this.bbox = new Value([0, 0, 0, 0]);
		this.byteSize = new Value(65536, true);
		this.count = new Value(0);
		this.geometryTypes = new Value([]);
		this.properties = new PropertiesValue([]);
		this.tags = new Value([]);
	}
}

class Value {

	constructor(value, estimated = false) {
		this.value = value;
		this.estimated = estimated;
	}
}

class PropertiesValue extends Value {

	constructor(value, estimated = false, searchable = "PARTIAL") {
		super(value, estimated);
		this.searchable = searchable;
	}
}

class PropertyInfo {

	constructor(key, count, searchable = false) {
		this.key = key;
		this.count = count;
		this.searchable = searchable;
	}
}

StatisticsResponse.Value = Value;
StatisticsResponse.PropertiesValue = PropertiesValue;
StatisticsResponse.Property = PropertyInfo;

module.exports = StatisticsResponse;

File: connectorSamples/xyz-airquality/src/StatisticsResponse.py
class StatisticsResponse:
    def __init__(self, bboxValue, byteSizeValue, countValue, geometryTypes, propertiesArray):
        self.type = "StatisticsResponse"
        self.bbox = {"value": bboxValue, "estimated": False}
        self.byteSize = {"value": byteSizeValue, "estimated": False}
        self.count = {"value": countValue, "estimated": False}
        self.geometryTypes = {"value": geometryTypes, "estimated": False}
        self.properties = {
            "value": propertiesArray,
            "estimated": False,
            "searchable": "PARTIAL"
        }
        self.tags = {
            "value": [],
            "estimated": False
        }

    def add_tags(self, key, count):
        self.tags["value"].append({
            "key": key,
            "count": count
        })

